scripts/create_vault.ts

```diff
@@ -16,47 +16,63 @@ import {
 } from "../sdk/src";
 
 const main = async () => {
-    const cluster: Cluster = "mainnet-beta";
-    //const cluster: Cluster = "devnet";
+    // const cluster: Cluster = "mainnet-beta";
+    const cluster: Cluster = "devnet";
     const connection = new Connection(
-        "https://solana-api.syndica.io/access-token/PBhwkfVgRLe1MEpLI5VbMDcfzXThjLKDHroc31shR5e7qrPqQi9TAUoV6aD3t0pg/rpc"
-        //"https://psytrbhymqlkfrhudd.dev.genesysgo.net:8899/"
+        // "https://solana-api.syndica.io/access-token/PBhwkfVgRLe1MEpLI5VbMDcfzXThjLKDHroc31shR5e7qrPqQi9TAUoV6aD3t0pg/rpc"
+        "https://psytrbhymqlkfrhudd.dev.genesysgo.net:8899/"
     );
     const wallet = Wallet.local();
+    const owner = wallet.payer;
     const provider = new Provider(connection, wallet, {
         commitment: "confirmed",
     });
 
-    //const reserveMint = NATIVE_MINT;
+    // const reserveMint = NATIVE_MINT;
     const reserveMint = new PublicKey(
         "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v"
     );
 
-    const solend = await SolendReserveAsset.load(
-        provider,
-        cluster,
-        reserveMint
-    );
-    const port = await PortReserveAsset.load(provider, cluster, reserveMint);
-    const jet = await JetReserveAsset.load(provider, cluster, reserveMint);
-
     const vaultClient = await VaultClient.initialize(
         provider,
         wallet,
         DeploymentEnvs.mainnet,
         reserveMint,
-        solend,
-        port,
-        jet,
-        wallet.publicKey,
+        owner,
         new PublicKey("jvUsXAgE2Gg92BbEBDAu7h5p8SEZpVjFqURJkzSsLNk"),
         {
             allocationCapPct: 60,
             rebalanceMode: { [RebalanceModes.calculator]: {} },
             strategyType: { [StrategyTypes.maxYield]: {} },
         }
     );
-    console.log(vaultClient.vaultId.toString());
+    console.log("Vauld ID: ", vaultClient.vaultId.toString());
+
+    try {
```

Comment:
I'm guessing these are empty catches because an error indicates that the asset doesn't exist?

1. Can you add a comment documenting that?
2. Can you add some logging in the catch? At least the error and a helpful message
3. Is it possible to be more specific on which errors are "acceptable" and which are not? What if there is an actual error in the rpc call to the program?

Reply:
To be very strict, only the exception of `SolendReserveAsset` should indicate pool unavailability, so we should capture exception of only `SolendReserveAsset`, not `initializeSolend`. 
However I'm catching the exception of both calls now because otherwise we need more code to decide whether we should call `initializeSolend` later, which is a bit too much IMO. We didn't catch initialization exceptions before anyway.